dev/com.ibm.ws.security.acme_fat/fat/src/com/ibm/ws/security/acme/fat/AcmeSwapDirectoriesTest.java

```diff
@@ -0,0 +1,488 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.security.acme.fat;
+
+import static junit.framework.Assert.assertEquals;
+import static junit.framework.Assert.assertFalse;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.math.BigInteger;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.cert.Certificate;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.StringTokenizer;
+
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TestName;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+import com.ibm.websphere.simplicity.log.Log;
+import com.ibm.ws.security.acme.docker.CAContainer;
+import com.ibm.ws.security.acme.docker.pebble.PebbleContainer;
+import com.ibm.ws.security.acme.utils.AcmeFatUtils;
+
+import componenttest.annotation.AllowedFFDC;
+import componenttest.annotation.CheckForLeakedPasswords;
+import componenttest.annotation.Server;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+/**
+ * Test for AcmeHistoricalFile. Ensure the file is created, updated, and
+ * causes refreshing of certificates.
+ */
+@RunWith(FATRunner.class)
+public class AcmeSwapDirectoriesTest {
+
+	@Server("com.ibm.ws.security.acme.fat.simple")
+	public static LibertyServer server;
+
+	protected static ServerConfiguration ORIGINAL_CONFIG;
+
+	/*
+	 * Domains that are configured and cleared before and after the class.
+	 */
+	private static final String[] DOMAINS_ALL = { "domain1.com", "domain2.com", "domain3.com" };
```

Comment:
I would expect us to only have 1 domain. We don't want to switch domains, otherwise the domain changing is what could be leading to the cert renew.

Reply:
We should not refresh the certificate if we remove a domain that is not the subject cn. I use this domain change to trigger a config update check. Normally, nothing would happen, but since we can't read from the ACME file, we should refresh the certificate.

Reply:
We will refresh the certificate if one of the domains is not in either the CN or the DNSNames in the SANs. I would just be careful changing domains, it may trigger a refresh if done in the wrong oder.

You might want to trigger a configuration change with something that can't possibly result in a new cert - like the polling times.